zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/Publisher.java

```diff
@@ -692,6 +703,25 @@ private static boolean publishDatasets(Collection collection) throws IOException
         return datasetsPublished;
     }
 
+    private static Future<Boolean> publishImages(Collection collection){
+        String collectionId = collection.getDescription().getId();
+
+        return pool.submit(() -> {
+            Boolean complete = false;
+            try {
+                imageServiceSupplier.getService().publishImagesInCollection(collection);
+                complete = true;
+            } catch (Exception e) {
+                error().data("collectionId", collectionId).data("publishing", true)
+                        .logException(e, "Exception publishing images via image API");
+
+                PostMessageField msg = new PostMessageField("Error", e.getMessage(), false);
+                collectionAlarm(collection, "Exception publishing images via image API", msg);
```

Comment:
I tried to avoid any error reporting within the core publishing code when it was originally put in. I guess there would be two schools of thought here:

1 - prevent any error reporting in the core publish code. Capture any exception and report it later once we're outside of the publish. Imagine a scenario where slack is down and the call to slack hangs. It would hold up the publish process. (This is where using `&&` in the point above may be useful)
2 - ensure that the image publishing code is sufficiently isolated from the old publish code, so that if it fails it has minimal impact on the existing publish. (This is where an isolated thread pool would be preferred)

Reply:
I agree with Carl's point here. 

I think the suggested way to handle exceptions thrown in a thread executor service  is to just let it be thrown. Once the tasks have completed you iterate over the list of futures calling  `future.get()` on each one. If a task failed `get` will throw an exception. You then catch this exception and handle it as appropriate.

You're error handling logic is spot on I think you just need to move entire catch block move it up to the catch block on line `135` where you are iterating over the futures.

Reply:
I have moved the catch block out of this future and to the code that completes the future in the calling method.